pandas/plotting/_matplotlib/core.py

```diff
@@ -1024,10 +1031,22 @@ def _make_plot(self):
         elif color is not None:
             c_values = color
         elif c_is_column:
-            c_values = self.data[c].values
+            if color_by_categorical:
+                c_values = self.data[c].cat.codes
+            else:
+                c_values = self.data[c].values
```

Comment:
Maybe slightly modify the logic here?

```
...
elif color_by_categorical:
  c_values = self.data[c].cat.codes
elif c_is_column:
  c_values = self.data[c].values
else:
  c_values = c
```

This allows one to eliminate the inner ``if-else`` statement.

Reply:
yes, good catch, thank you!